src/memory/AmdFlash.hh

```diff
@@ -262,14 +278,18 @@ public:
 	};
 
 	struct Sector {
-		size_t address;
-		power_of_two<size_t> size;
-		bool writeProtect;
+		size_t address = 0;
+		power_of_two<size_t> size = 1;
+		bool writeProtect = false;
+		std::optional<bool> erasePending = {};
```

Comment:
The meaning of this `optional<bool>` wasn't immediately clear to me.
I think you want the following 3 states, right?
* No erase operation is running at all.
* Erase is still running (or has not yet started).
* Erase operation on this sector has finished.

We should at least document more clearly which of these 3 maps on which state of the `optional<bool>`.
But I also wonder if an `enum` isn't better suited for this.